src/pages/GamePage/GamePage.tsx

```diff
@@ -166,6 +167,37 @@ const GamePage: React.FC<Props> = ({ currGameData, setCurrGameData }) => {
       });
     }
   };
+  const handleNextKeys = (index: number) => {
+    const targetKey = typingText.charAt(index) !== "\n" ? typingText.charAt(index) : "Enter";
+    const nextKeyArr: string[] = [];
+    if (keyData[targetKey].keyType === "default") nextKeyArr.push(targetKey);
+    else if (keyData[targetKey].keyType === "option") {
+      nextKeyArr.push("option");
+      Object.keys(keyData)
+        .filter(
+          (keyName) =>
+            keyData[keyName].keyType === "default" &&
+            keyData[keyName].position[0] === keyData[targetKey].position[0] &&
+            keyData[keyName].position[1] === keyData[targetKey].position[1]
+        )
+        .forEach((keyName) => {
+          nextKeyArr.push(keyName);
+        });
```

Comment:
```suggestion
      const k = Object.keys(keyData).find(
        (keyName) =>
          keyData[keyName].keyType === "default" &&
          keyData[keyName].position[0] === keyData[targetKey].position[0] &&
          keyData[keyName].position[1] === keyData[targetKey].position[1]
      );
      if(typeof k !== 'undefined') {
        nextKeyArr.push(k);
      } else {
        console.error('Invalid key');
      }
```

この処理ですが、keyDataの中から次に打つべきkeyを見つけるという処理をしていると思います。

条件に合致するkeyは必ず一つのはずなので、この場合はfindメソッドのほうが意味合いとしてはいいかもしれません。

また、filterにしろfindにしろ、返り値の検証を追加するといいと思います。

次のshiftのケースでも同様です。

Reply:
確かにfindの方が、無駄に全てのキーデータを検証しないため、計算量的にも良いですね！

返り値の検証も今後のデバッグ時に役立つと思うので良いと思います！